frontend/src/lib/components/TusUpload.svelte

```diff
@@ -1,43 +1,50 @@
-﻿<script lang="ts">
-  import { Upload, type DetailedError } from 'tus-js-client';
-  import { FormError, FormField, Button } from '$lib/forms';
-  import { env } from '$env/dynamic/public';
-  import { createEventDispatcher, onDestroy } from 'svelte';
-  import t from '$lib/i18n';
-
-  enum UploadStatus {
+﻿<script context="module" lang="ts">
+  export enum UploadStatus {
     NoFile = 'NoFile',
+    InvalidFile = 'InvalidFile',
     Ready = 'Ready',
     Error = 'Error',
     Uploading = 'Uploading',
-    Paused = 'Paused',
     Complete = 'Complete',
   }
+</script>
+
+<script lang="ts">
+  import { Upload, type DetailedError } from 'tus-js-client';
+  import { Button, FormError, FormField } from '$lib/forms';
+  import { env } from '$env/dynamic/public';
+  import { createEventDispatcher, onDestroy, onMount } from 'svelte';
+  import t from '$lib/i18n';
+  import IconButton from './IconButton.svelte';
 
   export let endpoint: string;
   export let accept: string;
-  export let uploadLabel: string = $t('tus.upload');
   export let inputLabel: string = $t('tus.select_file');
   export let inputDescription: string | undefined = undefined;
+  export let internalButton = false;
```

Comment:
I don't really like how I made the button external, but...I also don't want to spend lots of time on this.
If we need more file-inputs, then we can revisit it. I think this is acceptable for now.